farm/data_handler/samples.py

```diff
@@ -219,6 +219,11 @@ def chunk_into_passages(doc_offsets,
                         doc_text):
     """ Returns a list of dictionaries which each describe the start, end and id of a passage
     that is formed when chunking a document using a sliding window approach. """
+
+    assert doc_stride < passage_len_t, "doc_stride is longer than passage_len_t. This means that there will be gaps " \
```

Comment:
Let's switch this one to a logger.error() as it can be thrown at runtime rather than at init (like the other two)

Reply:
All errors thrown during the creation of samples and features are caught by a higher level try-catch (see processor._init_samples_in_baskets) which is why I thought I'd go for an assert statement here. I think it will be clearer to the user if its thrown as an Error here, then caught and printed at the higher level. Would you agree?

Reply:
Ok makes sense

Reply:
This more localized but caught assert I would prefer.

What about a more user friendly error msg since the user doesnt know about passage_len_t. 
What about describing passage len t: "Doc stride is longer than max_seq_len minus the tokens used for the question"